pkg/cache/snapshot_test.go

```diff
@@ -864,3 +864,457 @@ func TestSnapshotAddRemoveWorkload(t *testing.T) {
+			Obj(),
+	}
+
+	ctx := context.Background()
+	cl := utiltesting.NewClientBuilder().WithLists(&kueue.WorkloadList{Items: workloads}).Build()
+
+	cqCache := New(cl)
+	for _, flv := range flavors {
+		cqCache.AddOrUpdateResourceFlavor(flv)
+	}
+	for _, cq := range clusterQueues {
+		if err := cqCache.AddClusterQueue(ctx, cq); err != nil {
+			t.Fatalf("Couldn't add ClusterQueue to cache: %v", err)
+		}
+	}
+	wlInfos := make(map[string]*workload.Info, len(workloads))
+	for _, cq := range cqCache.clusterQueues {
+		for _, wl := range cq.Workloads {
+			wlInfos[workload.Key(wl.Obj)] = wl
+		}
+	}
+	initialSnapshot := cqCache.Snapshot()
+	initialCohortResources := initialSnapshot.ClusterQueues["lend-a"].Cohort.RequestableResources
+	cases := map[string]struct {
+		remove             []string
+		add                []string
+		want               Snapshot
+		enableLendingLimit bool
+	}{
+		"no-op remove add": {
+			remove:             []string{"/lend-a-1", "/lend-a-2", "/lend-a-3", "/lend-b-1"},
+			add:                []string{"/lend-a-1", "/lend-a-2", "/lend-a-3", "/lend-b-1"},
+			want:               initialSnapshot,
+			enableLendingLimit: true,
+		},
+		"remove all with lending limit": {
+			remove: []string{"/lend-a-1", "/lend-a-2", "/lend-a-3", "/lend-b-1"},
+			want: func() Snapshot {
+				cohort := &Cohort{
+					Name:                          "lend",
+					AllocatableResourceGeneration: 2,
+					RequestableResources:          initialCohortResources,
+					Usage: FlavorResourceQuantities{
+						"default": {corev1.ResourceCPU: 0},
+					},
+				}
+				return Snapshot{
+					ClusterQueues: map[string]*ClusterQueue{
+						"lend-a": {
+							Name:                          "lend-a",
+							Cohort:                        cohort,
+							Workloads:                     make(map[string]*workload.Info),
+							ResourceGroups:                cqCache.clusterQueues["lend-a"].ResourceGroups,
+							FlavorFungibility:             defaultFlavorFungibility,
+							AllocatableResourceGeneration: 1,
+							Usage: FlavorResourceQuantities{
+								"default": {corev1.ResourceCPU: 0},
+							},
+							GuaranteedQuota: FlavorResourceQuantities{
+								"default": {
+									corev1.ResourceCPU: 6_000,
+								},
+							},
+						},
+						"lend-b": {
+							Name:                          "lend-b",
+							Cohort:                        cohort,
+							Workloads:                     make(map[string]*workload.Info),
+							ResourceGroups:                cqCache.clusterQueues["lend-b"].ResourceGroups,
+							FlavorFungibility:             defaultFlavorFungibility,
+							AllocatableResourceGeneration: 1,
+							Usage: FlavorResourceQuantities{
+								"default": {corev1.ResourceCPU: 0},
+							},
+							GuaranteedQuota: FlavorResourceQuantities{
+								"default": {
+									corev1.ResourceCPU: 4_000,
+								},
+							},
+						},
+					},
+				}
+			}(),
+			enableLendingLimit: true,
+		},
+		"remove lend-a-2": {
```

Comment:
Plz make the test name more readable, like the intention you want to test here.

Reply:
Done.